File: java/src/test/java/leetcode/PalindromeNumberTest.java
// java/src/test/java/leetcode/PalindromeNumberTest.java

package leetcode;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class PalindromeNumberTest {

    @Test
    public void testSingleDigitNumbers() {
        assertTrue(PalindromeNumber.isPalindrome(0));
        assertTrue(PalindromeNumber.isPalindrome(7));
        assertTrue(PalindromeNumber.isPalindrome(9));
    }

    @Test
    public void testNegativeNumbers() {
        assertFalse(PalindromeNumber.isPalindrome(-121));
        assertFalse(PalindromeNumber.isPalindrome(-1));
        assertFalse(PalindromeNumber.isPalindrome(-101));
    }

    @Test
    public void testNumbersEndingInZero() {
        assertFalse(PalindromeNumber.isPalindrome(10));
        assertFalse(PalindromeNumber.isPalindrome(100));
        assertFalse(PalindromeNumber.isPalindrome(1000));
    }

    @Test
    public void testEvenLengthPalindromes() {
        assertTrue(PalindromeNumber.isPalindrome(11));
        assertTrue(PalindromeNumber.isPalindrome(1221));
        assertTrue(PalindromeNumber.isPalindrome(123321));
    }

    @Test
    public void testOddLengthPalindromes() {
        assertTrue(PalindromeNumber.isPalindrome(121));
        assertTrue(PalindromeNumber.isPalindrome(12321));
        assertTrue(PalindromeNumber.isPalindrome(1234321));
    }

    @Test
    public void testNonPalindromes() {
        assertFalse(PalindromeNumber.isPalindrome(12));
        assertFalse(PalindromeNumber.isPalindrome(123));
        assertFalse(PalindromeNumber.isPalindrome(1234));
    }

    @Test
    public void testLargerNumbers() {
        assertTrue(PalindromeNumber.isPalindrome(9999999));
        assertTrue(PalindromeNumber.isPalindrome(12344321));
        assertFalse(PalindromeNumber.isPalindrome(1234567));
        assertFalse(PalindromeNumber.isPalindrome(9876543));
    }

    @Test
    public void testPerformance1K() {
        long startTime = System.nanoTime();
        
        for (int i = 0; i < 1000; i++) {
            PalindromeNumber.isPalindrome(i);
        }
        
        long endTime = System.nanoTime();
        double durationMs = (endTime - startTime) / 1_000_000.0;
        
        System.out.printf("[1K iterations] Execution time: %.3fms%n", durationMs);
        assertTrue(durationMs < 100);
    }

    @Test
    public void testPerformance10K() {
        long startTime = System.nanoTime();
        
        for (int i = 0; i < 10000; i++) {
            PalindromeNumber.isPalindrome(i % 1000000);
        }
        
        long endTime = System.nanoTime();
        double durationMs = (endTime - startTime) / 1_000_000.0;
        
        System.out.printf("[10K iterations] Execution time: %.3fms%n", durationMs);
        assertTrue(durationMs < 200);
    }

    @Test
    public void testPerformance100K() {
        long startTime = System.nanoTime();
        
        for (int i = 0; i < 100000; i++) {
            PalindromeNumber.isPalindrome(i % 10000000);
        }
        
        long endTime = System.nanoTime();
        double durationMs = (endTime - startTime) / 1_000_000.0;
        
        System.out.printf("[100K iterations] Execution time: %.3fms%n", durationMs);
        assertTrue(durationMs < 1000);
    }
}

File: csharp/LeetCode/ArraysAndHashing/PalindromeNumberTest.cs
// csharp/LeetCode/ArraysAndHashing/PalindromeNumberTest.cs

using Xunit;
using System.Diagnostics;

namespace LeetCode.ArraysAndHashing
{
    public class PalindromeNumberTest
    {
        [Fact]
        public void TestSingleDigitNumbers()
        {
            Assert.True(PalindromeNumber.IsPalindrome(0));
            Assert.True(PalindromeNumber.IsPalindrome(7));
            Assert.True(PalindromeNumber.IsPalindrome(9));
        }

        [Fact]
        public void TestNegativeNumbers()
        {
            Assert.False(PalindromeNumber.IsPalindrome(-121));
            Assert.False(PalindromeNumber.IsPalindrome(-1));
            Assert.False(PalindromeNumber.IsPalindrome(-101));
        }

        [Fact]
        public void TestNumbersEndingInZero()
        {
            Assert.False(PalindromeNumber.IsPalindrome(10));
            Assert.False(PalindromeNumber.IsPalindrome(100));
            Assert.False(PalindromeNumber.IsPalindrome(1000));
        }

        [Fact]
        public void TestEvenLengthPalindromes()
        {
            Assert.True(PalindromeNumber.IsPalindrome(11));
            Assert.True(PalindromeNumber.IsPalindrome(1221));
            Assert.True(PalindromeNumber.IsPalindrome(123321));
        }

        [Fact]
        public void TestOddLengthPalindromes()
        {
            Assert.True(PalindromeNumber.IsPalindrome(121));
            Assert.True(PalindromeNumber.IsPalindrome(12321));
            Assert.True(PalindromeNumber.IsPalindrome(1234321));
        }

        [Fact]
        public void TestNonPalindromes()
        {
            Assert.False(PalindromeNumber.IsPalindrome(12));
            Assert.False(PalindromeNumber.IsPalindrome(123));
            Assert.False(PalindromeNumber.IsPalindrome(1234));
        }

        [Fact]
        public void TestLargerNumbers()
        {
            Assert.True(PalindromeNumber.IsPalindrome(9999999));
            Assert.True(PalindromeNumber.IsPalindrome(12344321));
            Assert.False(PalindromeNumber.IsPalindrome(1234567));
            Assert.False(PalindromeNumber.IsPalindrome(9876543));
        }

        [Fact]
        public void TestPerformance1K()
        {
            var sw = Stopwatch.StartNew();
            
            for (int i = 0; i < 1000; i++)
            {
                PalindromeNumber.IsPalindrome(i);
            }
            
            sw.Stop();
            Console.WriteLine($"[1K iterations] Execution time: {sw.Elapsed.TotalMilliseconds:F3}ms");
            
            Assert.True(sw.Elapsed.TotalMilliseconds < 100);
        }

        [Fact]
        public void TestPerformance10K()
        {
            var sw = Stopwatch.StartNew();
            
            for (int i = 0; i < 10000; i++)
            {
                PalindromeNumber.IsPalindrome(i % 1000000);
            }
            
            sw.Stop();
            Console.WriteLine($"[10K iterations] Execution time: {sw.Elapsed.TotalMilliseconds:F3}ms");
            
            Assert.True(sw.Elapsed.TotalMilliseconds < 200);
        }

        [Fact]
        public void TestPerformance100K()
        {
            var sw = Stopwatch.StartNew();
            
            for (int i = 0; i < 100000; i++)
            {
                PalindromeNumber.IsPalindrome(i % 10000000);
            }
            
            sw.Stop();
            Console.WriteLine($"[100K iterations] Execution time: {sw.Elapsed.TotalMilliseconds:F3}ms");
            
            Assert.True(sw.Elapsed.TotalMilliseconds < 1000);
        }
    }
}